### PR TITLE
[backend] Domain observable with hyphen/dash is incorrectly rejected V2 (#8927)

### PR DESCRIPTION
**Pull request by SarahBocognano**
### Proposed changes

* New regex format to NOT kill perf

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/8927
* https://github.com/OpenCTI-Platform/opencti/issues/7620

### CODE REVIEW

File: opencti-platform/opencti-graphql/tests/01-unit/utils/syntax-test.ts [hunk 1]
```diff
@@ -12,12 +12,12 @@ describe('Regex Pattern Tests', () => {
...
-    // expect('test-test.com').toMatch(domainChecker);
-    // expect('test-test.mytest.com').toMatch(domainChecker);
-    // expect('invalid_domain.12_3').not.toMatch(domainChecker);
-    // expect('invalid_domain').not.toMatch(domainChecker);
+    expect('test._mysubdomain.mydomain.com').toMatch(domainChecker);
+    expect('test_mysubdomain.domain.io').toMatch(domainChecker);
+    expect('test-test.com').toMatch(domainChecker);
+    expect('test-test.mytest.com').toMatch(domainChecker);
+    expect('invalid_domain.12_3').not.toMatch(domainChecker);
+    expect('invalid_domain').not.toMatch(domainChecker);
```
Comment (delemaf) [inline opencti-platform/opencti-graphql/tests/01-unit/utils/syntax-test.ts:24]: Theses 2 tests are failing

```js
    expect('invalid_domain.12_3').not.toMatch(domainChecker);
    expect('invalid_domain').not.toMatch(domainChecker);
```

Output:
```console
❯ tests/01-unit/utils/syntax-test.ts:20:39
     18|     expect('test-test.com').toMatch(domainChecker);
     19|     expect('test-test.mytest.com').toMatch(domainChecker);
     20|     expect('invalid_domain.12_3').not.toMatch(domainChecker);
       |                                       ^
     21|     expect('invalid_domain').not.toMatch(domainChecker);
     22|   });
```

File: opencti-platform/opencti-graphql/src/utils/syntax.js [hunk 2]
```diff
@@ -135,7 +135,7 @@ export const cleanupIndicatorPattern = (patternType, pattern) => {
 };
 
 export const systemChecker = /^\d{0,10}$/;
-export const domainChecker = /^(|(([\p{L}\d]|[\p{L}\d][\p{L}\d\p{Pd}_]*[\p{L}\d])\.)*([\p{L}\d]|[\p{L}\d][\p{L}\d\p{Pd}]*[\p{L}\d]))$/u;
+export const domainChecker = /^(?=.{1,253}$)(?!-)(?!_)([a-zA-Z0-9-_À-ÿ]{1,63}(\.[a-zA-Z0-9-_À-ÿ]{2,63})*)$/;
```
Comment (delemaf) [inline opencti-platform/opencti-graphql/src/utils/syntax.js]: This regex maybe fix failing test and fix perf issues, let me know if it works better for you 😃 
```suggestion
export const domainChecker = /^(?=.{1,253}$)(?!-)(?!.*--)([a-zA-Z0-9-_À-ÿ]{1,63}\.)+([a-zA-Z]{2,63})$/;
```
Should this regex match this domain : `invalid_domain.com` ? If yes maybe we could add a test to cover this case

**Comment by SarahBocognano** [comment]
Tests and CI errors have been resolved :)

**Review by aHenryJard — APPROVED** [review]
Tested locally looks good to me

**Pull request merged**